python/hail/representation/genotype.py

```diff
@@ -0,0 +1,278 @@
+from hail.java import *
+
+
+class Genotype(object):
+    """
+    An object that represents an individual's genotype at a genomic locus.
+
+    :param int gt: Genotype hard call
+    :param ad: allelic depth (1 element per allele including reference)
+    :type ad: list of int
+    :param int dp: total depth
+    :param int gq: genotype quality
+    :param pl: phred-scaled posterior genotype likelihoods (1 element per possible genotype)
+    :type pl: list of int
+    """
+
+    def __init__(self, gt, ad=None, dp=None, gq=None, pl=None):
+        """Initialize a Genotype object."""
+
+        jvm = Env.jvm()
+        gt = joption(gt)
+        if ad:
+            ad = jsome(jarray(jvm.int, ad))
+        else:
+            ad = jnone()
+        dp = joption(dp)
+        gq = joption(gq)
+        if pl:
+            pl = jsome(jarray(jvm.int, pl))
+        else:
+            pl = jnone()
```

Comment:
I would have left the attributes as python types.  Also, you don't want to make them attributes?  You're afraid someone will modify them?  Getter-only property is another option:

http://stackoverflow.com/questions/14594120/python-read-only-property